packages/components/src/DateTimePickers/InputDateTimePicker/date-extraction.test.js

```diff
@@ -43,44 +44,98 @@ describe('Date extraction', () => {
 		});
 	});
 
-	describe('extractPartsFromDateTime', () => {
-		it('should return empty parts on invalid date', () => {
+	describe('extractParts', () => {
+		it('should return empty parts on undefined value', () => {
 			// given
-			const invalidDate = 'lol';
 			const options = {
 				dateFormat: 'YYYY-MM-DD',
 				useTime: true,
 				useSeconds: true,
 			};
 
 			// when
-			const parts = extractPartsFromDateTime(invalidDate, options);
+			const parts = extractParts(undefined, options);
 
 			// then
 			expect(parts).toEqual({
 				date: undefined,
 				time: { hours: '', minutes: '', seconds: '' },
-				datetime: 'lol',
+				datetime: undefined,
 				textInput: '',
 			});
 		});
 
-		it('should return empty parts on undefined date', () => {
+		it('should return parts from timestamp', () => {
+			// given
+			const date = new Date(2015, 8, 15, 12, 58, 22);
+			const timestamp = date.getTime();
+			const options = { dateFormat: 'YYYY-MM-DD' };
+
+			// when
+			const parts = extractParts(timestamp, options);
+
+			// then
+			expect(parts).toEqual({
+				date: new Date(2015, 8, 15),
+				datetime: date,
+				textInput: '2015-09-15',
+				time: { hours: '00', minutes: '00', seconds: '00' },
+			});
+		});
+
+		it('should return parts from Date', () => {
+			// given
+			const validDate = new Date(2015, 8, 15, 12, 58, 22);
+			const options = { dateFormat: 'YYYY-MM-DD' };
+
+			// when
+			const parts = extractParts(validDate, options);
+
+			// then
+			expect(parts).toEqual({
+				date: new Date(2015, 8, 15),
+				datetime: validDate,
+				textInput: '2015-09-15',
```

Comment:
so is it 8 or 9?

Reply:
In date api 8 is the month index, that is used for september.
It is transformed to reflext `09` in the input to be human understandable.

Reply:
right !
it's been a while since I worked with datetime